### DISCUSSION TITLE
[Styled Components] How can we leverage the new SSG methods to export static style sheets at build time? 

### DISCUSSION THREAD

**Discussion by heyitsarpit** [vercel/next.js#11073] [Help]
The current way to do server side rendering in Next js requires creating a _document.jsx file in the pages directory, it uses Next's getInitialProps method which runs at each request.

Can we use the new functionality in 9.3 to create our <styles> at build time.

This would allow the app to render for the first time without any flickers(the app initially loads without any styles and CSS is injected later).

**Comment by timneutkens** [comment]
>  Next js requires creating a _document.jsx file in the pages directory, it uses Next's getInitialProps method which runs at each request.

`_document` getInitialProps **does not affect**:
- automatic static optimization
- `getStaticProps`

It's only ran when the render occurs, meaning that with `getStaticProps` and automatic static optimization it runs at `next build` time.

Basically it just works in a backwards compatible way.

**Reply by happy-unicorn** [reply]
@timneutkens Instead of getInitialProps, does getStaticProps in _document actually work?

**Reply by timneutkens** [reply]
No, you don't have to change anything in _document as of right now. As said it does not affect anything.
